docs/collecting-data/collecting-from-own-applications/javascript-trackers/browser-tracker/browser-tracker-v3-reference/advanced-usage/using-an-id-service/index.md

```diff
@@ -0,0 +1,154 @@
+- If `spIdService` and `sp` are both missing, then it generates a new ID in the `spIdService` and `sp` cookies with the same unique identifier generation algorithm with the Snowplow pipeline, currently UUID v4.
+
+### Code examples 
+
+Below we showcase a couple of code samples for ID service API endpoints:
+
+<Tabs groupId="id-service" queryString>
+<TabItem value="nextjs" label="Next.js TypeScript" default>
+
+```ts
+export default function handler(
+  req: NextApiRequest,
+  res: NextApiResponse<Data>
+) {
+  const { sp, spIdService } = req.cookies;
+  const expiration = timeInTwoYears();
+  const networkUserId = sp || spIdService || uuidv4();
+
+  res.setHeader("Set-Cookie", [
+     /* 
+       1. The domain will be the (e)TLD of the website. In this example it is snowplow.io
+       2. The cookie header attributes should have exactly the same values as the ones set on the collector configuration.
+      */
+    `spIdService=${networkUserId}; Expires=${expiration.toUTCString()}; Domain=snowplow.io; Path=/; Secure; SameSite=None; httpOnly;`,
+    `sp=${networkUserId}; Expires=${expiration.toUTCString()}; Domain=snowplow.io; Path=/; Secure; SameSite=None; httpOnly;`,
+  ]);
+
+  /* The response needs to return a 200 (OK) status code. */
+  res.status(200).json({ Ok: 200 });
+}
+```
+
+</TabItem>
+<TabItem value="php" label="PHP">
+
+```php
+function id_service() {
+  $sp = $_COOKIE['sp'] ?? null;
+  $spIdService = $_COOKIE['spIdService'] ?? null;
+  $expiration = time() + (2 * 365 * 24 * 60 * 60);
+  $networkUserId = $sp ?? $spIdService ?? gen_uuid4();
+
+  /* 
+    1. The domain will be the (e)TLD of the website. In this example it is snowplow.io
+    2. The cookie header attributes should have exactly the same values as the ones set on the collector configuration.
+  */
+  $cookie_options = array(
+    'expires' => $expiration,
+    'path' => '/',
+    'domain' => 'snowplow.io',
+    'secure' => true,
+    'httponly' => true,
+    'samesite' => 'None',
+  );
+  setcookie('sp', $networkUserId, $cookie_options);
+  setcookie('spIdService', $networkUserId, $cookie_options);
+
+  return json_encode(['Ok' => 200]);
+}
+```
+
+</TabItem>
+
+</Tabs>
+
+## Using the ID service on the Snowplow browser tracker
+
+When the ID service has been deployed on a system with the same resolved IP as the main document, the tracker can then be configured to orchestrate the required ID service API calls.
+
+This process is opt-in by using the `idService` option during tracker initialization:
+
+```tsx
+newTracker('sp1', 'c.customer.com', {
+  idService: "customer.com/id-service-endpoint",
+  /* ...Rest of the tracker options */
+ });
+```
+
+When the tracker detects this option it will send an HTTP request during initialization on this endpoint to have the service set the requires identifiers before sending any event.
```

Comment:
```suggestion
When the tracker detects this option it will send an HTTP request during initialization on this endpoint to have the service set the required identifiers before sending any event.
```